server/errors.go

```diff
@@ -76,6 +76,8 @@ func handleServerError(writer http.ResponseWriter, err error) {
 	switch err := err.(type) {
 	case *RouterMissingParamError, *RouterParsingError, *json.SyntaxError, *NoBodyError:
 		respErr = responses.SendError(writer, err.Error())
+	case *json.UnmarshalTypeError:
+		respErr = responses.SendError(writer, "bad type in json data")
```

Comment:
LGTM overall, is bad type the only error this can mean?

Reply:
I guess, because it's called UnmarhsalTypeError :)

Reply:
@Sergey1011010 I mean, why not the err.Error()?

Reply:
@Bee-lee err.Error() would reveal some data about internal types. The project is open-sourced, but still. @tisnik what do you think?

Reply:
@Sergey1011010 yeah I'm not sure in this case, when I was doing the refactoring, I replaced the messages and err.Errors for the InternalServerErrors, which can definitely reveal some stuff, so I see what you mean, but I don't know, this is still a client error

Reply:
IMHO it is ok, logs are stored quite securely :-)

Reply:
it is about types, not actual data, right?

Reply:
@tisnik I mean that if we return err.Error() to the user it would show message which would contain info about about type, something like:
```
cannot unmarshal array into Go struct field TypeName.FieldName of type string
```